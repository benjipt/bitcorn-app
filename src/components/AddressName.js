import React from 'react'

export default function AddressName({ address }) {
    return (
        <div className="mt-1 ps-4">
            <h4>{ address }</h4>
        </div>
    )
}
