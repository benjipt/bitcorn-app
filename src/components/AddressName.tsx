interface AddressNameProps {
  address: string;
}

export default function AddressName({ address }: AddressNameProps) {
  return (
    <div className='mt-1 ps-4' data-testid='AddressName-1'>
      <h4>{address}</h4>
    </div>
  );
}
