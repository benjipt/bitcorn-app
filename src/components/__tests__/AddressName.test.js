import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import AddressName from '../AddressName'

afterEach(() => {
    cleanup()
})

test('should render AddressName component with address prop', () => {
    const address = 'Jilly'
    render(<AddressName address={ address } />)
    const AddressNameComponent = screen.getByTestId('AddressName-1')
    expect(AddressNameComponent).toBeInTheDocument()
    expect(AddressNameComponent).toHaveTextContent('Jilly')
})

test('matches snapshot', () => {
    const address = 'Jilly'
    const tree = renderer.create(<AddressName address={ address } />).toJSON()
    expect(tree).toMatchSnapshot()
})