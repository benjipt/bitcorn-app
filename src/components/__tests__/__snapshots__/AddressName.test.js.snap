// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="mt-1 ps-4"
  data-testid="AddressName-1"
>
  <h4>
    Jilly
  </h4>
</div>
`;
